plugins/org.python.pydev.parser/src/org/python/pydev/parser/visitors/NodeUtils.java

```diff
@@ -1548,10 +1548,9 @@ public static TypeInfo getTypeForClassDefAttribute(String attributeWithoutSelf,
             stmtType stmtType = body[i];
             if (stmtType instanceof Assign) {
                 Assign assign = (Assign) stmtType;
-                if (assign.type != null) {
-                    String representationString = getRepresentationString(assign.type);
-                    if (attributeWithoutSelf.equals(representationString)) {
-                        return new TypeInfo(assign.value);
+                if (attributeWithoutSelf.equals(getRepresentationString(assign.targets[0]))) {
+                    if (assign.type != null) {
```

Comment:
You can just move this to the block below (which already checks if assign.targets != null and its len).